[Storage] Set live test timeout to 20 minutes

mikeharder opened the pull request:
This change was already validated in another PR we are using to test fixes.  Here's a build showing the 20-minute timeout in action.  So I'm not planning to trigger another run of storage live tests for this PR.

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=208734&view=ms.vss-test-web.build-test-results-tab

mikeharder committed:
[Storage] Set live test timeout to 20 minutes

tg-msft reviewed (approved):
Is it possible to apply this at the class level and have it inherit?  If so, we might be able to just put this on `StorageTestBase` (or even `ClientTestBase` if we want this for all Track 2 tests).

mikeharder commented:
> Is it possible to apply this at the class level and have it inherit? If so, we might be able to just put this on `StorageTestBase` (or even `ClientTestBase` if we want this for all Track 2 tests).

I know this attribute can be applied at the assembly, class, or method level.  I'm not sure if inheritance is handled correctly but I will try it.  I would put it on `ClientTestBase` since we do want this for all Track 2 tests.  If any test happens to need longer than 20 minutes they can override this attribute with a larger value.

mikeharder commented:
@tg-msft: Unfortunately the attribute is not inherited:

https://github.com/nunit/nunit/blob/edb0abc185af3e780e835bbb76318154e0b13091/src/NUnitFramework/framework/Attributes/TimeoutAttribute.cs#L36

If we wanted to apply this to all test assemblies, we could create a common `AssemblyInfo.cs` that all test assemblies compile.

weshaggard commented:
This does seem like a good thing to make common for all our test projects.

mikeharder commented:
> This does seem like a good thing to make common for all our test projects.

Unfortunately NUnit has a pretty bad bug where `TimeoutAttribute` breaks `Assert.Inconclusive()` (and `Assert.Pass()` though I doubt we use this in our tests):

https://github.com/nunit/nunit/issues/3282

For now, we're only going to enable this for Storage to avoid issues in other test projects which might use `Assert.Inconclusive()`.

mikeharder committed:
Add LiveOnly attribute to Assert.Inconclusive() tests

- Workaround for NUnit bug "TimeoutAttribute makes Assert.Inconclusive() count as failure" (nunit/nunit#3282)

mikeharder committed:
Set per-test timeout in shared AssemblyInfo.cs

mikeharder committed:
Remove unnecessary AssemblyInfo.cs

mikeharder committed:
Add missing using
